File: automation-samples/playwright-framework/src/pages/DashboardPage.ts
import { expect, Page } from "@playwright/test";

export class DashboardPage {
  constructor(private readonly page: Page) {}

  async expectLoaded() {
    await expect(this.page.getByRole("heading", { name: "Dashboard" })).toBeVisible();
    await expect(this.page.getByTestId("account-summary-card")).toBeVisible();
  }

  async expectNotificationsPanelVisible() {
    await expect(this.page.getByTestId("notifications-panel")).toBeVisible();
  }

  async signOut() {
    await this.page.getByRole("button", { name: "Open profile menu" }).click();
    await this.page.getByRole("menuitem", { name: "Sign out" }).click();
  }
}

File: automation-samples/selenium-framework/src/test/java/com/sangram/qa/pages/DashboardPage.java
package com.sangram.qa.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class DashboardPage {
    private final WebDriver driver;

    private final By title = By.cssSelector("[data-testid='dashboard-title']");
    private final By accountSummary = By.cssSelector("[data-testid='account-summary-card']");
    private final By profileMenu = By.cssSelector("[data-testid='profile-menu']");
    private final By signOut = By.cssSelector("[data-testid='sign-out']");

    public DashboardPage(WebDriver driver) {
        this.driver = driver;
    }

    public WebElement title() {
        return driver.findElement(title);
    }

    public WebElement accountSummary() {
        return driver.findElement(accountSummary);
    }

    public void signOut() {
        driver.findElement(profileMenu).click();
        driver.findElement(signOut).click();
    }
}
